source/connection/flow/tcp/tcp_flow.cpp

```diff
@@ -316,8 +322,12 @@ void TcpFlow::process_ack(Packet ack, std::size_t confirm_count) {
 
 Packet TcpFlow::generate_data_packet(PacketNum packet_num) {
     Packet packet;
-    m_flag_manager.set_flag(packet.flags, m_packet_type_label,
-                            PacketType::DATA);
+    packet.flags = get_flag_manager();
+    auto result = packet.flags.set_flag(m_packet_type_label, PacketType::DATA);
+    if (!result.has_value()) {
+        LOG_ERROR(std::move(result.error()));
+    }
```

Comment:
Looks like same code lines occurs many times, so you need method if `StdExpected::apply_to_errror` that applies given function to error if unexpected value stored. Please, add it
```suggestion
    packet.flags.set_flag(m_packet_type_label, PacketType::DATA).apply_to_error(
        [](const std::string& err) {LOG_ERROR(err);}
    );
```